lib/logthrdest/logthrdestdrv.c

```diff
@@ -1034,13 +1043,38 @@ log_threaded_dest_driver_set_max_retries_on_error(LogDriver *s, gint max_retries
   self->retries_on_error_max = max_retries;
 }
 
+static guint
+_get_worker_key_hash(LogThreadedDestDriver *self, LogMessage *msg)
```

Comment:
This should go to lib/templates instead. We already have two users (one in the logscheduler one here)

Reply:
I think the whole `is_literal`, `is_trivial` optimization should go under log_template_format instead, and then this becomes a one-liner.

Reply:
That function always requires a buffer to copy the data into. Avoiding that is the whole purpose of is_literal and is_trivial.

Reply:
I added log_template_hash().